### PR TITLE
feat: support sending emails in parallel and in sequential

### PR DESCRIPTION
**Pull request by shojisoma**
Closes https://github.com/SAP/cloud-sdk-backlog/issues/820

this converts the sequential email sending process into a parallel one.

### CODE REVIEW

**Review by FrankEssenberger — APPROVED** [review]
LGTM - did you check that the E2E tests are green?

File: packages/mail-client/src/mail-client.ts [hunk 1]
```diff
@@ -196,21 +196,30 @@ async function sendMailWithNodemailer<T extends MailOptions>(
...
+      );
+      return transport.sendMail({
+        ...mailOptionsFromDestination,
+        ...mailOption
+      });
+    }
+  );
+
+  const response: MailResponse[] = await Promise.all(promises).then(
+    responces => {
```
Comment (FrankEssenberger) [inline packages/mail-client/src/mail-client.ts:212]: [req] typo

Comment (shojisoma) [reply packages/mail-client/src/mail-client.ts:212]: done

**Review by jjtang1985 — CHANGES_REQUESTED** [review]
As the OnPrem (cloud worked) e2e tests failed, I would not merge this.
Potential follow up would be adding an option so users can decide the strategy (parallel or in sequential).

**Comment by FrankEssenberger** [comment]
> As the OnPrem (cloud worked) e2e tests failed, I would not merge this. Potential follow up would be adding an option so users can decide the strategy (parallel or in sequential).

Was the fail surely related to the code change so do they work with the `main` codeline?

**Comment by jjtang1985** [comment]
> Was the fail surely related to the code change so do they work with the main codeline?

The app was re-deployed after updating the SDK Core manually by copy/pasting the mail-client dist folder from the PR branch.

**Comment by jjtang1985** [comment]
I'll create a new PR

**Pull request closed without merging**